x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/LookupJoin.java

```diff
@@ -71,4 +75,14 @@ protected NodeInfo<Join> info() {
             config().rightFields()
         );
     }
+
+    @Override
+    public void postAnalysisVerification(Failures failures) {
+        super.postAnalysisVerification(failures);
+        if (right() instanceof EsRelation esr) {
+            if (esr.concreteIndices().contains(esr.indexPattern()) == false) {
```

Comment:
Also: Could there be weird edge cases where the name of the index contains e.g. an escaped wildcard character, so that an index pattern _is_ being used, but coincidentally, it also matches a concrete index? There's no escaping in index names, right? Per the docs it doesn't seem so: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html

I understand this is as edge case-y as it gets; but since avoiding aliases is required to ensure permissions are respected correctly, we better try to think about every possible scenario.